owlapy/owl_reasoner.py

```diff
@@ -80,36 +49,34 @@ def __init__(self, ontology: AbstractOWLOntology, *, class_cache: bool = True,
             """
         super().__init__(ontology)
         assert isinstance(ontology, Ontology)
-        self._world = ontology._world
-        self._ontology = ontology
-        self.class_cache = class_cache
-        self._property_cache = property_cache
-        self._negation_default = negation_default
-        self._sub_properties = sub_properties
-        self.__warned = 0
+        self._world: owlready2.World = ontology._world
+        self._ontology: Ontology = ontology
+        self.class_cache: bool = class_cache
+        self._property_cache: bool = property_cache
+        self._negation_default: bool = negation_default
+        self._sub_properties: bool = sub_properties
+        self.__warned: int = 0
         self._init()
 
-    def _init(self, cache_size=128):
-
-        individuals = self._ontology.individuals_in_signature()
```

Comment:
I see that we are removing the line where we get the individuals once and we are adding individual retrieval on line 606, 647, 767, 803 which will be killing the algo in large KGs.

Reply:
Yes, i just replaced ind_set with the individuals_in_signature() method when ever it is used. 
Ig an advantage is that this is only computed if it is really need and not always on initialisation

Reply:
Let's get the correctness done first, then focus on trading of the memory with runtimes. 
It is my mistake that I should have not accept the PR based on caching.